main.go

```diff
@@ -103,25 +103,37 @@ func run(l *log.Logger, args arguments) error {
 		l.Println(string(output))
 	}
 
-	var out io.Writer
-	if args.outputFile == "-" {
-		out = os.Stdout
+	var input io.Reader
+	if args.templateFile == "-" {
+		input = os.Stdin
 	} else {
-		var err error
-		out, err = os.Create(args.outputFile)
+		f, err := os.Open(args.templateFile)
 		if err != nil {
 			return err
 		}
+
+		defer f.Close()
+
+		input = f
 	}
 
-	var input io.Reader
-	if args.templateFile == "-" {
-		input = os.Stdin
+	var out io.Writer
```

Comment:
switched the order so we fail without writing anything if there's a borked input file.